PanelPal/main.py

```diff
@@ -1,43 +1,37 @@
-"""
-Main entry point for the PanelPal toolkit, which provides functionalities
-for querying and generating BED files for genomic panels.
-
-The script offers several primary subcommands:
-
-1. **check-panel**:
-    This subcommand allows users to check panel information based on the panel ID.
-    It queries panel details, such as name and version, from the PanelApp API.
-
-2. **generate-bed**:
-    This subcommand generates a BED file for a specified genomic panel.
-    It requires the panel ID, version, and genome build to generate the appropriate BED file.
-
-3. **compare-panel-versions**
-    This subcommand allows users to compare two versions of the same panel ID.
-    It requires the panel ID, two version numbers, and can take a filter of gene status.
-
-Usage examples:
-    - To check panel information for a specific panel ID:
-        $ PanelPal check-panel --panel_id R59
-
-    - To generate a BED file for a specific panel, version, and genome build:
-        $ PanelPal generate-bed --panel_id R59 --panel_version 4 --genome_build GRCh38
-
-    - To query the gene differences between two versions of a panel:
-        $ PanelPal compare-panel-versions --panel_id R21 --versions 1.0 2.2 --status_filter green
-"""
-
 import argparse
 from check_panel import main as check_panel_main
 from generate_bed import main as generate_bed_main
 from compare_panel_versions import main as compare_panel_versions_main
 from compare_panel_versions import validate_panel
 
 
+def print_help():
+    """Print custom help message for PanelPal."""
+    help_message = """
+PanelPal: A toolkit for panelapp queries
+version: 1.0.0 
+
+Available Commands:
+    check-panel             Check panel information for a given panel ID.
+                            Example: PanelPal check-panel --panel_id R59
+
+    generate-bed            Generate a BED file for a genomic panel. Requires the
+                            panel ID, panel version, and genome build.
+                            Example: PanelPal generate-bed --panel_id R59 --panel_version 4 --genome_build GRCh38
+
+    compare-panel-versions  Compare two versions of a genomic panel. Requires
+                            the panel ID and two version numbers. Optionally, filter by gene status.
+                            Example: PanelPal compare-panel-versions --panel R21 --versions 1.0 2.2 --status_filter green
+    --help, -h              Prints this help message
+    """
+    print(help_message)
+
+
 def main():
-    """Main function which gathers arguments and passes them to the relevant PanelPal command"""
+    """Main function which gathers arguments and passes them to the relevant PanelPal command."""
     parser = argparse.ArgumentParser(
-        description="PanelPal: A toolkit for panelapp queries"
+        description="PanelPal: A toolkit for querying and generating genomic panel BED files.",
+        epilog="For more details, visit https://yourprojecturl.com"
```

Comment:
Put actual repository link

Reply:
Addressed in commit [23629c9](https://github.com/PatrickWeller/PanelPal/pull/88/commits/23629c9d06f57d449c9e14bb0725b388dffb8d96)